cmake/FindCharm.cmake

```diff
@@ -62,15 +66,54 @@
 #  Set CHARM_ROOT before calling find_package to a path to add an additional
 #  search path, e.g.,
 #
-#  Usage:
+#  Usage
+#  -----
 #
 #  set(CHARM_ROOT "/path/to/custom/charm") # prefer over system
 #  find_package(Charm)
 #  if(CHARM_FOUND)
 #    # Link executables with the charmc wrapper
-#    STRING(REGEX REPLACE "<CMAKE_CXX_COMPILER>" "${CHARM_COMPILER}"
+#    STRING(REGEX REPLACE "<CMAKE_CXX_COMPILER>" "${CHARM_LINKER}"
 #           CMAKE_CXX_LINK_EXECUTABLE "${CMAKE_CXX_LINK_EXECUTABLE}")
 #  endif()
+#
+#  Assumptions
+#  -----------
+#
+#  An implicit assumption is a high-level of compatability between the
+#  compilers used to build charm++ and the compilers used to build cmake
+#
+#  We may be able to explicitly test this with `charmc -print-building-blocks`
+#  function (we could probably use this to extract some info rather than
+#  perform a bunch of the tests in this file)
+#
+#  Additional Notes
+#  ----------------
+#
+#  In case this approach ever breaks in the future...
+#
+#  It's frankly surprising that modifying CMAKE_CXX_LINK_EXECUTABLE to employ
+#  charmc as a linker (after a call to find_package) acheives the desired
+#  result. Precise details about modifying this variable is not particularly
+#  well documented by CMake. Starting in CMake version 3.29, there are now
+#  other "more official ways" to control the choice of linker.
+#
+#  While there does not appear to be exact documentation describing exactly how
+#  CMAKE_CXX_LINK_EXECUTABLE, based on most of CMake's generic semantics, there
+#  is reason to be skeptical of this kind of approach.
+#  - In general, cmake expects variables like these to be set in a "toolchain
+#    file" (essentially as early in the cmake startup as possible).
+#  - Unfortunately, this approach is a undesirable since users need to craft
+#    their toolchain files ahead of time (and they can't rely upon the
+#    find_package command)
+#
+#  The convential approach (that we should probably adopt if the current
```

Comment:
The parenthetical in this sentence doesn't actually end - you might want to add it?